src/chain_sync/network_context.rs

```diff
@@ -348,12 +351,19 @@ where
             .duration_since(req_pre_time)
             .unwrap_or_default();
         match res {
-            Ok(Ok(Ok(bs_res))) => {
-                // Successful response
-                peer_manager.log_success(peer_id, res_duration).await;
-                debug!("Succeeded: ChainExchange Request to {peer_id}");
-                Ok(bs_res)
-            }
+            Ok(Ok(Ok(bs_res))) => match bs_res.status {
+                ChainExchangeResponseStatus::Success
+                | ChainExchangeResponseStatus::PartialResponse => {
+                    peer_manager.log_success(peer_id, res_duration).await;
+                    debug!("Succeeded: ChainExchange Request to {peer_id}");
+                    Ok(bs_res)
+                }
+                _ => {
+                    peer_manager.log_failure(peer_id, res_duration).await;
+                    debug!("Failed: ChainExchange Request to {peer_id}");
+                    Err(format!("{:?}: {}", bs_res.status, bs_res.message))
```

Comment:
I'm not convinced we should return an error here. A "i-dont-have-the-block" response is still a valid response.